apps/api/src/donations/donations.service.ts

```diff
@@ -76,10 +76,21 @@ export class DonationsService {
     ]
   }
 
-  async listDonations(): Promise<Donation[]> {
-    return await this.prisma.donation.findMany({
-      orderBy: [{ person: { firstName: 'asc' } }, { createdAt: 'desc' }],
-    })
+  /**
+   * Lists all donations
+   * @param campaignId (Optional) Filter by campaign id
+   */
+  async listDonations(campaignId?: string): Promise<Donation[]> {
+    if (campaignId) {
+      return await this.prisma.donation.findMany({
+        where: { targetVault: {campaign: {id: campaignId}} },
+        orderBy: [{ createdAt: 'desc' }],
+      })
+    } else {
+      return await this.prisma.donation.findMany({
+        orderBy: [{ createdAt: 'desc' }],
+      })
+    }
```

Comment:
Just tested: Here there is no need of the if condition, the where clause will return all if the id is undefined.
so leave only this code:
```
return await this.prisma.donation.findMany({
        where: { targetVault: {campaign: {id: campaignId}} },
        orderBy: [{ createdAt: 'desc' }],
      })
```

Reply:
not exactly - if the value is an empty string then there will be an exception. While the routing middleware correctly converts empty string to undefined, this business logic can be reused from anywhere in the code, where such conversion will not be available

Reply:
I think we're good to continue as is with the `campaignId` param